### ISSUE TITLE
Simplify prugeExpired

### ISSUE THREAD

**Issue by axman6** [hverr/haskell-cache#10]
It looks like purgeExpired's implementation is just a more complicated was of running `HashMap.filter (not . isExpired now)`, which would likely be much more efficient than repeatedly reading and writing the TVar for each key.

**Comment by hverr** [comment]
You are correct! I currently don't have time to make any changes, but I'd welcome a PR. otherwise, it might take a while.

**Comment by axman6** [comment]
I'll see what I can do tonight.
